### PR TITLE
PS-3788: (PS 5.7 partially dynamic @version variable using support of @version_suffix 

### PR DESCRIPTION
**Pull request by inikep**
Tested at https://jenkins.percona.com/view/5.7/job/mysql-5.7-param/1576/

### CODE REVIEW

File: sql/sys_vars.h [hunk 1]
```diff
@@ -840,34 +840,54 @@ class Sys_var_charptr: public sys_var
 
 class Sys_var_version : public Sys_var_charptr
 {
+private:
+  char withsuffix[SERVER_VERSION_LENGTH];
```
Comment (laurynas-biveinis) [inline sql/sys_vars.h:844]: s/char/uchar/g ?

File: sql/sys_vars.h [hunk 2]
```diff
@@ -840,34 +840,54 @@ class Sys_var_charptr: public sys_var
...
+
+    memcpy(withsuffix, *version_ptr, version_ptr_len);
+    memcpy(withsuffix + version_ptr_len, *suffix_ptr, suffix_ptr_len);
+    withsuffix[suffix_ptr_len + version_ptr_len] = 0;
+
+    return reinterpret_cast<uchar*> (&withsuffix_ptr);
   }
 };
 
-
```
Comment (laurynas-biveinis) [inline sql/sys_vars.h]: Spurious change

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

File: sql/sys_vars.h (hunk 1 above)
Comment (inikep) [reply sql/sys_vars.h:844]: I do not understand. What do you mean?
`char` is on purpose as is minimizes casts for `memcpy` and other.

Comment (laurynas-biveinis) [reply sql/sys_vars.h:844]: Ack, OK then. My suggestion was in order to remove some of casts with uchar, please keep whichever results in fewer casts.

**Pull request merged**